src/main/java/Deadline.java

```diff
@@ -10,4 +10,9 @@ public Deadline(String description, String by) {
     public String toString() {
         return "[D]" + super.toString() + " (by: " + by + ")";
     }
+
+    @Override
+    public String PopulateSaveData() {
+        return "D | " + (this.isDone ? 1 : 0) + " | " + this.description + " | " + this.by;
+    }
```

Comment:
Change your method declaration to match the naming standard, which means that this should be changed to populateSaveData() instead.